SUPPORTED_APIS.md

```diff
@@ -2,31 +2,122 @@
+| `ETH` | `eth_getLogs` | NOT IMPLEMENTED | Returns an array of all logs matching a given filter object |
+| `ETH` | `eth_getProof` | NOT IMPLEMENTED | Returns the details for the account at the specified address and block number, the account's Merkle proof, and the storage values for the specified storage keys with their Merkle-proofs |
+| `ETH` | `eth_getStorageAt` | NOT IMPLEMENTED | Returns the value from a storage position at a given address |
+| `ETH` | `eth_getTransactionByBlockHashAndIndex` | NOT IMPLEMENTED | Returns information about a transaction by block hash and transaction index position |
+| `ETH` | `eth_getTransactionByBlockNumberAndIndex` | NOT IMPLEMENTED | Returns information about a transaction by block number and transaction index position |
+| `ETH` | `eth_getTransactionByHash` | NOT IMPLEMENTED | Returns the information about a transaction requested by transaction hash |
+| `ETH` | `eth_getTransactionCount` | NOT IMPLEMENTED | Returns the number of transactions sent from an address |
+| `ETH` | `eth_getTransactionReceipt` | NOT IMPLEMENTED | Returns the receipt of a transaction by transaction hash |
+| `ETH` | `eth_getUncleByBlockHashAndIndex` | NOT IMPLEMENTED | Returns information about a uncle of a block by hash and uncle index position |
+| `ETH` | `eth_getUncleByBlockNumberAndIndex` | NOT IMPLEMENTED | Returns information about a uncle of a block by hash and uncle index position |
+| `ETH` | `eth_getUncleCountByBlockHash` | NOT IMPLEMENTED | Returns the number of uncles in a block from a block matching the given block hash |
+| `ETH` | `eth_getUncleCountByBlockNumber` | NOT IMPLEMENTED | Returns the number of uncles in a block from a block matching the given block hash |
+| `ETH` | `eth_getWork` | NOT IMPLEMENTED | Returns: An Array with the following elements 1: DATA, 32 Bytes - current block header pow-hash 2: DATA, 32 Bytes - the seed hash used for the DAG. 3: DATA, 32 Bytes - the boundary condition ("target"), 2^256 / difficulty |
+| `ETH` | `eth_hashrate` | NOT IMPLEMENTED | Returns the number of hashes per second that the node is mining with |
+| `ETH` | `eth_maxPriorityFeePerGas` | NOT IMPLEMENTED | Returns a `maxPriorityFeePerGas` value suitable for quick transaction inclusion |
+| `ETH` | `eth_mining` | NOT IMPLEMENTED | Returns true if client is actively mining new blocks |
+| `ETH` | `eth_newBlockFilter` | NOT IMPLEMENTED | Creates a filter in the node, to notify when a new block arrives |
+| `ETH` | `eth_newFilter` | NOT IMPLEMENTED | Creates a filter object, based on filter options, to notify when the state changes (logs) |
+| `ETH` | `eth_newPendingTransactionFilter` | NOT IMPLEMENTED | Creates a filter in the node, to notify when new pending transactions arrive |
+| `ETH` | `eth_protocolVersion` | NOT IMPLEMENTED | Returns the current ethereum protocol version |
+| `ETH` | `eth_sendRawTransaction` | NOT IMPLEMENTED | Creates new message call transaction or a contract creation for signed transactions |
```

Comment:
This is `IMPLEMENTED`: https://github.com/matter-labs/era-test-node/blob/main/src/node.rs#L678

Reply:
This one's a little confusing. The code points to bytes but I send this:

```bash
POST http://localhost:8011
content-type: application/json

{
    "jsonrpc": "2.0",
    "id": "1",
    "method": "eth_sendRawTransaction",
    "params": ["bogus"]
}
```

And returns:

```bash
{
  "jsonrpc": "2.0",
  "error": {
    "code": -32602,
    "message": "Invalid params: invalid value: string \"bogus\", expected 0x prefix."
  },
  "id": "1"
}
```

and sending:

```bash
POST http://localhost:8011
content-type: application/json

{
    "jsonrpc": "2.0",
    "id": "1",
    "method": "eth_sendRawTransaction",
    "params": ["0x0000"]
}
```

returns:

`socket hang up` on VSCode and `thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: UnknownTransactionFormat', src/node.rs:688:82` on logs for node

Reply:
Sending `"bogus"` is interpreted as a string and not `bytes` in Rust. If you prefix with `0x` than it'll read as bytes in RPC.

Yeah, the node doesn't gracefully handle errors when you send random bytes that aren't a real transaction. This is the work @dutterbutter is working on right now as part of this upcoming release.

Reply:
Great! I pushed a success `eth_call` example, and tried using that for this function too, but with the same error message as a result sadly. Here's what I tried: 

```js
const obj = {
        "to": "0x36615Cf349d7F6344891B1e7CA7C72883F5dc049",
        "data": "0x0000",
        "from": "0xa61464658AfeAf65CccaaFD3a512b69A83B77618",
        "gas": "0x0000",
        "gasPrice": "0x0000",
        "value": "0x0000",
        "nonce": "0x0000"
};

function utf8ToHex(str) {
  return '0x' + Array.from(str).map(c =>
    c.charCodeAt(0) < 128 ? c.charCodeAt(0).toString(16) :
    encodeURIComponent(c).replace(/\%/g,'').toLowerCase()
  ).join('');
}

console.log(utf8ToHex(JSON.stringify(obj)));
```

to:

```bash
###
POST http://localhost:8011
content-type: application/json

{
    "jsonrpc": "2.0",
    "id": "1",
    "method": "eth_sendRawTransaction",
    "params": ["0x7b22746f223a22307833363631354366333439643746363334343839314231653743413743373238383346356463303439222c2264617461223a22307830303030222c2266726f6d223a22307861363134363436353841666541663635436363616146443361353132623639413833423737363138222c22676173223a22307830303030222c226761735072696365223a22307830303030222c2276616c7565223a22307830303030222c226e6f6e6365223a22307830303030227d"]
}
```

Reply:
Have modified it to `IMPLEMENTED` on the table now  👍

Reply:
I think `eth_sendRawTransaction` is still showing as `NOT IMPLEMENTED`

Reply:
Keen eye, added:) @MexicanAce